core/worker.cc

```diff
@@ -105,7 +105,13 @@ static void resume_worker(int wid) {
   }
 }
 
-void resume_all_workers() {
+
+/*!
+ * Attach orphan TCs to workers. Note this does not ensure optimal placement.
+ * This method can only be called when all workers are paused.
+ */
+void attach_orphans() {
+  assert(!is_any_worker_running());
```

Comment:
This should be `CHECK`. We compile with DNEDBUG, so this won't have the intended behavior unless you do a debug build